packages/jest-matchers/src/index.js

```diff
@@ -47,7 +47,10 @@ function makeThrowingMatcher(
         message = message();
       }
 
-      throw new Error(message);
+      const error = new Error(message);
+      // Remove this function from the stack trace frame.
+      Error.captureStackTrace(error, throwingMatcher);
```

Comment:
Well, this is not the only place that can throw an error. Any test can throw whatever they want at any time.

I do agree however that once we are done replacing *all* matchers, we should revisit formatFailureMessages.